Source/HeaderableTableDataDisplayManager.swift

```diff
@@ -0,0 +1,178 @@
+//  ReactiveDataDisplayManager
+//
+//  Created by Ivan Smetanin on 24/09/2017.
+//  Copyright © 2017 Александр Кравченков. All rights reserved.
+//
+
+import Foundation
+
+/// Protocol for work with cell.
+public protocol HeaderableTableDataDisplayManager: class {
+
+    /// Add generator of header for section.
+    func addSectionHeaderGenerator(_ generator: HeaderGenerator)
+
+    /// This method is used to add the new cell generator.
+    ///
+    /// - Parameters:
+    ///   - generator: Generator which should be added.
+    ///   - header: Header in which is added to generator, if nil generator
+    /// will be added to the last header.
+    ///   - needRegister: Pass true if needed to register the cell.
+    func addCellGenerator(_ generator: TableCellGenerator, toHeader header: HeaderGenerator?, needRegister: Bool)
+
+    /// Set tableView for current manager
+    func setTableView(_ tableView: UITableView)
+}
+
+public class BaseHeaderableTableDataDisplayManager: NSObject, HeaderableTableDataDisplayManager {
+
+    // MARK: - Events
+
+    /// Called if table scrolled
+    public var scrollEvent = BaseEvent<UITableView>()
+
+    // MARK: - Fileprivate properties
+
+    fileprivate var cellGenerators: [[TableCellGenerator]]
+    fileprivate var sectionHeaderGenerators: [HeaderGenerator]
+    fileprivate weak var tableView: UITableView?
+    fileprivate let estimatedHeight: CGFloat
+
+    // MARK: - Initialization and deinitialization
+
+    public init(estimatedHeight: CGFloat = 40) {
+        self.estimatedHeight = estimatedHeight
+        self.cellGenerators = [[TableCellGenerator]]()
+        self.sectionHeaderGenerators = [HeaderGenerator]()
+        super.init()
+    }
+
+    /// Set TableView to current adapter.
+    ///
+    /// - Parameter tableView: new TableView
+    public func setTableView(_ tableView: UITableView) {
+        self.tableView = tableView
+        self.tableView?.delegate = self
+        self.tableView?.dataSource = self
+    }
+}
+
+// MRK: - Generator actions
+
+public extension BaseHeaderableTableDataDisplayManager {
+
+    /// Added new header for section generator.
+    ///
+    /// - Parameter generator: new generator.
+    public func addSectionHeaderGenerator(_ generator: HeaderGenerator) {
+        self.sectionHeaderGenerators.append(generator)
+    }
+
+    public func addCellGenerator(_ generator: TableCellGenerator, toHeader header: HeaderGenerator? = nil, needRegister: Bool = true) {
+        if needRegister {
+            self.tableView?.registerNib(generator.identifier)
+        }
+        if self.cellGenerators.count != self.sectionHeaderGenerators.count || sectionHeaderGenerators.isEmpty {
+            self.cellGenerators.append([TableCellGenerator]())
+        }
+
+        if let header = header {
+            if let index = self.sectionHeaderGenerators.index(where: { $0 === header }) {
+                self.cellGenerators[index].append(generator)
+            }
```

Comment:
Maybe here and below need add fatalError with description that header doesn't find or doesn't exist